tensorboard/components/tf_backend/router.ts

```diff
@@ -88,9 +89,9 @@ export function setRouter(router: Router): void {
 
 function createProdPath(pathPrefix: string, path: string,
     ext: string, params?: URLSearchParams): string {
-  const url = new URL(window.location.origin);
   // Use URL to normalize pathPrefix with leading slash and without.
-  url.pathname = pathPrefix + path;
+  const maybeRelativePath = pathPrefix + path;
+  const url = new URL(maybeRelativePath, window.location.href);
```

Comment:
Do we need to use the actual window.location here?  Could we just pass "http://localhost" or even "http://ignored" instead?  That would make it clearer that we're only dealing with relative URL paths.

Reply:
The point of using `window.location.href` is that if you’ve navigated to
<http://localhost:6006/some-prefix/> then that prefix will be included
in the eventual URL. Doesn’t this break if you use `"http://localhost"`
instead of `window.location.href`? Am I missing something?

Reply:
But maybeRelativePath already includes the configured pathPrefix.  Unless the idea here was to make TensorBoard automatically detect the pathPrefix from the href, I don't think it's necessary to get it from window.location.href.

And actually, I don't think any path information is retained from the second parameter to URL at all, from testing it just now - it only extracts the origin anyway:

```js
> new URL('prefix/foo', 'http://xyz')
URL {href: "http://xyz/prefix/foo", origin: "http://xyz", protocol: "http:", username: "", password: "", …}
> new URL('prefix/foo', 'http://xyz/')
URL {href: "http://xyz/prefix/foo", origin: "http://xyz", protocol: "http:", username: "", password: "", …}
> new URL('prefix/foo', 'http://xyz/foo')
URL {href: "http://xyz/prefix/foo", origin: "http://xyz", protocol: "http:", username: "", password: "", …}
> new URL('/prefix/foo', 'http://xyz/foo')
URL {href: "http://xyz/prefix/foo", origin: "http://xyz", protocol: "http:", username: "", password: "", …}
```

Reply:
Note
```
> new URL('prefix/foo', 'http://xyz/foo')
URL {href: "http://xyz/prefix/foo", origin: "
> new URL('prefix/foo', 'http://xyz/foo/')
URL {href: "http://xyz/foo/prefix/foo", origin: "
```

Reply:
Your tests are correct but nonexhaustive. If you’re on `http://xyz/foo`
and navigate to `prefix/bar`, you will visit `http://xyz/prefix/bar`.
You need to include a trailing slash to force an additional path
component:

```js
> String(new URL("y/z", "http://foo/"))
"http://foo/y/z"
> String(new URL("y/z", "http://foo/bar/"))
"http://foo/bar/y/z"
> String(new URL("y/z", "http://foo/bar/baz/"))
"http://foo/bar/baz/y/z"
```

> But maybeRelativePath already includes the configured pathPrefix

Yes, but the `pathPrefix` is always `"data"`, right? (Admittedly, this
naming is very misleading.)

Reply:
> The router used to entirely ignore the path prefix (because it only
> operated on relative paths). With this change, the use of
> window.location.href will bake it into the final pathname as an
> absolute URL path (starting with /), but it's still not explicitly
> referenced.

Yes, exactly.

> So really we should probably change everything here back to dataDir
> instead of pathPrefix. 

Fine with me. It’s worth noting that the value of this argument is in
fact always `"data"` in public code, except for tests of the router
itself.